axis.yaml

```diff
@@ -0,0 +1,14 @@
+CUDA_VER:
+  - "11.2.2"
+  - "11.8.0"
+PYTHON_VER:
+  - "3.9"
+  - "3.10"
+LINUX_VER:
+  - "ubuntu20.04"
+  - "ubuntu22.04"
```

Comment:
Do we need both `ubuntu` versions?

Didn't we talk about just supporting the latest `ubuntu` version and omitting the OS entirely from the resulting tag name?

Logically it seems like we should use a single OS (the latest Ubuntu) for the same reasons that we dropped CentOS/RockyLinux image variants.

Reply:
Yes we did discuss using a single Ubuntu. Unfortunately, at the time we did not consider that we had also committed to supporting CUDA 11.2.x (eg we only [dropped 11.4 & 11.5](https://docs.rapids.ai/notices/rsn0027/)), but CUDA 11.2 is not supported on Ubuntu 22.04.

So I decided to add Ubuntu 20.04 for the CUDA 11.2 support.

Reply:
We could build Ubuntu 20.04 only for CUDA 11.2 and Ubuntu 22.04 for CUDA 11.8+. And then drop the `LINUX_VER` from the tag. Basically, the CUDA version you pull dictates the Ubuntu version.

This would be less transparent, but simpler in a way.

Reply:
> We could build Ubuntu 20.04 only for CUDA 11.2 and Ubuntu 22.04 for CUDA 11.8+. And then drop the `LINUX_VER` from the tag. Basically, the CUDA version you pull dictates the Ubuntu version.
> 
> This would be less transparent, but simpler in a way.

I'm in favor of this.

It's less complicated for our end users. They don't need to worry about what Ubuntu version they're pulling, just the CUDA & Python version.

The consequence of not doing this will mean that everyone who consumes our images will have to keep track of which Ubuntu version they need to specify for each CUDA version. It also means that logic will need to be incorporated in other areas across RAPIDS like the release selector on our website.

Reply:
I'm weakly in favor of retaining the Ubuntu version in the tag. For folks building on top of RAPIDS containers, it can make a pretty big difference if you have 20.04 vs. 22.04. Having that be implicit in the CUDA version is not ideal and doesn't communicate helpful info to users. I always search/browse tags directly on Docker Hub, so I like to know what I'm getting.

Reply:
> I'm weakly in favor of retaining the Ubuntu version in the tag. For folks building on top of RAPIDS containers, it can make a pretty big difference if you have 20.04 vs. 22.04. Having that be implicit in the CUDA version is not ideal and doesn't communicate helpful info to users. I always search/browse tags directly on Docker Hub, so I like to know what I'm getting.

I'd rather us include the mapping between CUDA version & Ubuntu version in the GitHub/DockerHub README's than in the image tags.

Something like this...

```markdown
The Ubuntu version included in each image is the latest supported by each CUDA version.

See the table below for details.

| CUDA Version        | Ubuntu Version       |
| ------------------- | -------------------- |
| <=11.2              | 20.04                |
| >11.2               | 22.04                |
```

IMO, we should only include the items that we explicitly paramterize (Python & CUDA version) in the image tags.

Reply:
I'm leaning towards the implicit Ubuntu version and documenting it as AJ suggests.

This is not uncommon for docker tags. For example, `python:3.10` is Debian 12 based and an alias of `python:3.10-bookworm` but you could use `python:3.10-bullseye` or `python:3.10-alpine` to get a different base OS. But the default tag essentially picks the "works-best-for-most" OS.

I don't think we want to build the entire matrix of base OSes, but at least there is some precedent for selecting the "best fit" OS automatically.

Reply:
Forgive me, I might be missing something here. If I am reading this right, we aren't going to be shipping containers with other base OSes like Rocky Linux or CentOS (only ubuntu20.04 and ubuntu22.04 are listed here) It looks like `rapidsai/rapidsai` has tags for `centos7` and so forth, but the overhauled images won't support RHEL-like images? If we're only shipping Ubuntu, then the calculus changes for me, and I'm okay with making the OS more of a hidden parameter.

Reply:
That's correct, Bradley. We aren't shipping `centos`/`rockylinux` variants for the overhauled images.

Ok, so it seems like we're all on the same page now then.

Let's nix those OS strings.